phases/18-ethics-safety-alignment/21-fairness-criteria-group-individual-counterfactual/code/main.py

```
@@ -0,0 +1,134 @@
+        x0 = random.gauss(0.8 * y, 1.0)
+        x1 = random.gauss(-0.3 + a * 0.5, 1.0)
+        data.append(([x0, x1, float(a)], y, a))
+    return data
+
+
+def train(data, steps: int = 200, lr: float = 0.1, sample_weights=None) -> list[float]:
+    w = [0.0, 0.0, 0.0]
+    b = 0.0
+    for _ in range(steps):
+        random.shuffle(data)
+        for idx, (x, y, a) in enumerate(data):
```

Ruff (0.15.11)

[warning] Loop control variable `a` not used within loop body

Rename unused `a` to `_a`

(B007)